### PR TITLE
Shade okhttp and okio in jaxrs-clients

### PR DESCRIPTION
**Pull request by uschi2000**
Required by https://github.com/palantir/http-remoting/issues/431

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/http-remoting/432)

### CODE REVIEW

**Comment by uschi2000** [comment]
@ellisjoe

**Comment by uschi2000** [comment]
@markelliot

**Comment by uschi2000** [comment]
any comments/concerns?

File: jaxrs-clients/build.gradle [hunk 1]
```diff
@@ -54,6 +58,8 @@ shadowJar {
     relocate('zipkin', 'httpremoting.shaded.zipkin')
     relocate('org.joda.time', 'httpremoting.shaded.org.joda.time')
     relocate('okio', 'httpremoting.shaded.okio')
+    relocate('okhttp3', 'httpremoting.shaded.okhttp3')
+    relocate('com.palantir.remoting2.tracing.okhttp3', 'httpremoting.shaded.com.palantir.remoting2.tracing.okhttp3')
```
Comment (ellisjoe) [inline jaxrs-clients/build.gradle:62]: why does this need to get relocated?

**Comment by uschi2000** [comment]
I don't want to copies of this on the classpath with the same name: one in
the shadowJar and one from others depending on this library.

**Review by ellisjoe — APPROVED** [review]

**Comment by j-baker** [comment]
want to flag that this PR causes some serious breaks internally - it's binary incompatible with previous releases. I've linked you some code - but basically there are widely used internal libraries that rely on FailoverFeignTarget taking an okhttp3.OkHttpClient and not a `httpremoting.shaded.okhttp3.OkHttpClient`

**Pull request merged**